patches/third_party/webrtc/002-crop_video_to_prevent_freezes.patch

```diff
@@ -0,0 +1,75 @@
+diff --git a/video/vie_encoder.cc b/video/vie_encoder.cc
+index 3a848f852..825451084 100644
+--- a/video/vie_encoder.cc
++++ b/video/vie_encoder.cc
+@@ -494,6 +494,20 @@ void ViEEncoder::ReconfigureEncoder() {
+       encoder_config_.video_stream_factory->CreateEncoderStreams(
+           last_frame_info_->width, last_frame_info_->height, encoder_config_);
+ 
++  // TODO(ilnik): If configured resolution is significantly less than provided,
++  // e.g. because there are not enough SSRCs for all simulcast streams,
++  // signal new resolutions via SinkWants to video source.
++
++  // Stream dimensions may be not equal to given because of a simulcast
++  // restrictions.
++  int highest_stream_width = static_cast<int>(streams.back().width);
++  int highest_stream_height = static_cast<int>(streams.back().height);
++  // Dimension may be reduced to be, e.g. divisible by 4.
++  RTC_CHECK_GE(last_frame_info_->width, highest_stream_width);
++  RTC_CHECK_GE(last_frame_info_->height, highest_stream_height);
++  crop_width_ = last_frame_info_->width - highest_stream_width;
++  crop_height_ = last_frame_info_->height - highest_stream_height;
++
+   VideoCodec codec;
+   if (!VideoCodecInitializer::SetupCodec(encoder_config_, settings_, streams,
+                                          nack_enabled_, &codec,
+@@ -690,12 +704,34 @@ void ViEEncoder::EncodeVideoFrame(const VideoFrame& video_frame,
+   }
+   TraceFrameDropEnd();
+ 
++  VideoFrame out_frame(video_frame);
++  // Crop frame if needed.
++  if (crop_width_ > 0 || crop_height_ > 0) {
++    int cropped_width = video_frame.width() - crop_width_;
++    int cropped_height = video_frame.height() - crop_height_;
++    rtc::scoped_refptr<I420Buffer> cropped_buffer =
++        I420Buffer::Create(cropped_width, cropped_height);
++    // TODO(ilnik): Remove scaling if cropping is too big, as it should never
++    // happen after SinkWants signaled correctly from ReconfigureEncoder.
++    if (crop_width_ < 4 && crop_height_ < 4) {
++      cropped_buffer->CropAndScaleFrom(
++          *video_frame.video_frame_buffer(), crop_width_ / 2, crop_height_ / 2,
```

Comment:
This and line 47 are slightly modified to remove ->ToI420() from the original change, because the I420Buffer interface changed in the interim.